### PR TITLE
[#618] chore(docker): support downloads latest hadoop archives and mirror url

### PR DESCRIPTION
**Pull request by advancedxy**
### What changes were proposed in this pull request?
1. support download latest hadoop archive such as hadoop-2.10.2
2. add `--apache-mirror` for build.sh

### Why are the changes needed?
Fix: #618 

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Manually tested.

### CODE REVIEW

File: deploy/kubernetes/docker/build.sh [hunk 1]
```diff
@@ -72,9 +79,10 @@ fi
 
 HADOOP_FILE=hadoop-${HADOOP_VERSION}.tar.gz
 HADOOP_URL=https://archive.apache.org/dist/hadoop/core/hadoop-${HADOOP_VERSION}/${HADOOP_FILE}
-echo "HADOOP_URL is $HADOOP_URL"
+LATEST_HADOOP_URL=${APACHE_MIRROR}/hadoop/core/hadoop-${HADOOP_VERSION}/${HADOOP_FILE}
```
Comment (kaijchen) [inline deploy/kubernetes/docker/build.sh:82]: Maybe `MIRROR_HADOOP_URL`?

Comment (kaijchen) [reply deploy/kubernetes/docker/build.sh:82]: OK, I got it after looking into the issue, maybe call this `HADOOP_URL`, and change line 81 to `ARCHIVE_HADOOP_URL`

File: deploy/kubernetes/docker/build.sh [hunk 2]
```diff
@@ -51,6 +54,10 @@ while (( "$#" )); do
     --help)
       exit_with_usage
       ;;
+    --apache-mirror)
+      APACHE_MIRROR="$2"
```
Comment (wangao1236) [inline deploy/kubernetes/docker/build.sh:58]: Perhaps we can decide not to replace the default value if the value passed in is empty?

Comment (wangao1236) [reply deploy/kubernetes/docker/build.sh:58]: Ok

**Review by wangao1236 — APPROVED** [review]

**Review by kaijchen — APPROVED** [review]
LGTM, thanks @advancedxy and @wangao1236.

**Pull request merged**